File: scala/unwrittenfun/minecraft/unwrittenblocks/UnwrittenBlocks.scala
package unwrittenfun.minecraft.unwrittenblocks

import cpw.mods.fml.common.Mod
import unwrittenfun.minecraft.unwrittenblocks.handlers.{GuiHandler, ConfigHandler, PacketHandler}
import cpw.mods.fml.common.network.NetworkMod
import cpw.mods.fml.common.Mod.EventHandler
import unwrittenfun.minecraft.unwrittenblocks.items.UBItems
import cpw.mods.fml.common.event.{FMLInitializationEvent, FMLPreInitializationEvent}
import unwrittenfun.minecraft.unwrittenblocks.blocks.UBBlocks
import cpw.mods.fml.common.registry.GameRegistry
import unwrittenfun.minecraft.unwrittenblocks.worldgen.GenPleather


/**
 * Mod: UnwrittenBlocks
 * Author: UnwrittenFun
 * License: Minecraft Mod Public License (Version 1.0.1)
 */

@Mod(modid = ID, name = NAME, version = VERSION, modLanguage = "scala")
@NetworkMod(channels = Array(CHANNEL), clientSideRequired = true, serverSideRequired = false, packetHandler = classOf[PacketHandler])
object UnwrittenBlocks {
  @EventHandler def preInit(event: FMLPreInitializationEvent) {
    ConfigHandler init event.getSuggestedConfigurationFile

    UBItems.registerItems()
    UBBlocks.registerBlocks()
  }

  @EventHandler def load(event: FMLInitializationEvent) {
    UBItems.registerNames()
    UBBlocks.registerNames()

    UBBlocks.registerRecipes()
    UBItems.registerRecipes()

    UBBlocks.registerTileEntities()

    GameRegistry registerWorldGenerator new GenPleather()

    new GuiHandler()
  }
}

File: src/main/java/unwrittenfun/minecraft/unwrittenblocks/common/UnwrittenBlocks.java
package unwrittenfun.minecraft.unwrittenblocks.common;

import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.SidedProxy;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.network.NetworkRegistry;
import cpw.mods.fml.common.registry.GameRegistry;
import net.minecraft.creativetab.CreativeTabs;
import org.apache.logging.log4j.Logger;
import unwrittenfun.minecraft.unwrittenblocks.client.creativetab.UBCreativeTab;
import unwrittenfun.minecraft.unwrittenblocks.common.blocks.BlockRegister;
import unwrittenfun.minecraft.unwrittenblocks.common.handlers.GuiHandler;
import unwrittenfun.minecraft.unwrittenblocks.common.items.ItemRegister;
import unwrittenfun.minecraft.unwrittenblocks.common.network.NetworkRegister;
import unwrittenfun.minecraft.unwrittenblocks.common.worldgen.WorldGenPleather;


/**
 * Project: UnwrittenBlocks
 * Author: UnwrittenFun
 * Created: 25/08/2014.
 */

@Mod(modid = ModInfo.MOD_ID, name = ModInfo.NAME, version = ModInfo.VERSION)
public class UnwrittenBlocks {
  public static Logger logger;

  @SidedProxy(clientSide = "unwrittenfun.minecraft.unwrittenblocks.client.ClientProxy", serverSide = "unwrittenfun.minecraft.unwrittenblocks.common.CommonProxy")
  public static CommonProxy proxy;

  @Mod.Instance
  public static UnwrittenBlocks instance;

  public static CreativeTabs creativeTabUB;

  @Mod.EventHandler
  public void preInit(FMLPreInitializationEvent event) {
    logger = event.getModLog();

    creativeTabUB = new UBCreativeTab();

    BlockRegister.registerBlocks();
    ItemRegister.registerItems();

    BlockRegister.registerRecipes();
    ItemRegister.registerRecipes();

    GameRegistry.registerWorldGenerator(new WorldGenPleather(), 10);

    proxy.registerRenderers();

    NetworkRegister.initMessages();

    NetworkRegistry.INSTANCE.registerGuiHandler(this, new GuiHandler());
  }
}
